app_test.go

```diff
@@ -5,56 +5,242 @@
 package kintone
 
 import (
-	"bytes"
+	"crypto/tls"
 	"fmt"
+	"io"
 	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
 	"strings"
 	"testing"
 	"time"
 )
 
-const KINTONE_DOMAIN = "YOUR_DOMAIN"
-const KINTONE_USER = "YOUR_KINTONE_USER"
-const KINTONE_PASSWORD = "YOUR_KINTONE_PASSWORD"
-const KINTONE_API_TOKEN = "YOUR_API_TOKEN"
+const (
+	KINTONE_DOMAIN         = "localhost:8088"
+	KINTONE_USERNAME       = "test"
+	KINTONE_PASSWORD       = "test"
+	KINTONE_APP_ID         = 1
+	KINTONE_API_TOKEN      = "1e42da75-8432-4adb-9a2b-dbb6e7cb3c6b"
+	KINTONE_GUEST_SPACE_ID = 1
+	AUTH_HEADER_TOKEN      = "X-Cybozu-API-Token"
+	AUTH_HEADER_PASSWORD   = "X-Cybozu-Authorization"
+)
 
-func newApp(appID uint64) *App {
-	return &App{
-		Domain:   KINTONE_DOMAIN,
-		User:     KINTONE_USER,
-		Password: KINTONE_PASSWORD,
-		AppId:    appID,
+func createServerTest(mux *http.ServeMux) (*httptest.Server, error) {
+	ts := httptest.NewUnstartedServer(mux)
+	listen, err := net.Listen("tcp", KINTONE_DOMAIN)
+
+	if err != nil {
+		return nil, err
+	}
+
+	ts.Listener.Close()
+	ts.Listener = listen
+	ts.StartTLS()
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	return ts, nil
+}
+
+func createServerMux() (*http.ServeMux, error) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/k/v1/record.json", handleResponseGetRecord)
+	mux.HandleFunc("/k/v1/records.json", handleResponseGetRecords)
+	mux.HandleFunc("/k/v1/record/comments.json", handleResponseGetRecordsComments)
+	mux.HandleFunc("/k/v1/file.json", handleResponseUploadFile)
+	mux.HandleFunc("/k/v1/record/comment.json", handleResponseRecordComments)
+	mux.HandleFunc("/k/v1/records/cursor.json", handleResponseRecordsCursor)
+	mux.HandleFunc("/k/v1/form.json", handleResponseForm)
+	mux.HandleFunc("/k/guest/1/v1/form.json", handleResponseForm)
+	return mux, nil
+}
+
+// header check
+func checkAuth(response http.ResponseWriter, r *http.Request) {
+	authPassword := r.Header.Get(AUTH_HEADER_PASSWORD)
+	authToken := r.Header.Get(AUTH_HEADER_TOKEN)
+	if authPassword == "" && authToken == "" {
```

Comment:
This logic should be checked:
- Check authentication by password from header must be equal the base64(username:password).
- Check authentication by password & authentication by token should be not same times, user can use authentication by password or token.